internal/syscallfs/readfs_test.go

```diff
@@ -11,22 +11,50 @@ import (
 	"github.com/tetratelabs/wazero/internal/testing/require"
 )
 
-func TestReadFS_MkDir(t *testing.T) {
+func TestNewReadFS(t *testing.T) {
 	tmpDir := t.TempDir()
-	testFS := NewReadFS(Adapt("/", hackFS(tmpDir)))
 
-	err := testFS.Mkdir("mkdir", fs.ModeDir)
+	// Doesn't double-wrap file systems that are already read-only
```

Comment:
this is better than before where we accidentally wrapped things that didn't need it